server/idapiProxy.ts

```diff
@@ -28,8 +28,8 @@ interface IdapiFetchOptions {
 }
 
 interface CookiesWithToken {
-	SC_GU_U: string;
-	[key: string]: string;
+	SC_GU_U?: string;
+	[key: string]: string | undefined;
```

Comment:
it doesn't feel like having a cookie with no value makes sense, is that really a thing?

Reply:
having checked, it seems like you must have a value (the `=` is mandatory) but it may have an empty string as the value (there may be 0 or more characters after the `=`)

https://datatracker.ietf.org/doc/html/rfc6265#:~:text=cookie%2Dpair%20%20%20%20%20%20%20%3D%20cookie%2Dname%20%22%3D%22%20cookie%2Dvalue

So it might be that the upstream thing needs tweaking rather than allowing undefined?

Reply:
ah I see that you made another change while I was trying to understand!  But I'd actually worked out that it would be undefined if the underlying cookie value was not a recognised format for the cookie-parser library.
This means that the "cookie" value returned by this library could be
- not present at all if it wasn't set at all, or
- undefined if it was set to something not starting with `"j:"`